sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestBindBean.java

```diff
@@ -68,4 +77,59 @@ public void testBindBeanPrefix() {
         @SqlQuery("select name from something where id = :id")
         String getName(long id);
     }
+
+    @Test
+    public void testNoArgumentFactoryRegisteredForProperty() {
+        handle.execute("create table beans (id integer, value_type varchar)");
+
+        assertThatThrownBy(() -> handle.attach(BeanDao.class).insert(new Bean(1, ValueType.valueOf("foo"))))
+                .hasMessageContaining("No argument factory registered");
+
+        handle.registerArgument(new ValueTypeArgumentFactory());
```

Comment:
I think it's better to split this test to two: one for checking the error message and the other one for the successful execution with the registered mapper.